traffic_ops/traffic_ops_golang/role/roles.go

```diff
@@ -177,12 +179,16 @@ func (role *TORole) Update() (error, error, int) {
 	if userErr != nil || sysErr != nil {
 		return userErr, sysErr, errCode
 	}
+
 	// TODO cascade delete, to automatically do this in SQL?
-	userErr, sysErr, errCode = role.deleteRoleCapabilityAssociations(role.ReqInfo.Tx)
-	if userErr != nil || sysErr != nil {
-		return userErr, sysErr, errCode
+	if role.Capabilities != nil && *role.Capabilities != nil {
```

Comment:
do you need to check for len of *role.Capabilities?

Reply:
Not to fix the bug.

Sounds like it might be a good idea, though, as it's probably doing extra work otherwise

Reply:
Oh, no actually, you still need to do that. If the Role had associated capabilities and the user updated it to have none, you still need to delete those associations.

Reply:
ahh - makes sense 👍